db/c.cc

```diff
@@ -1341,14 +1341,19 @@ void rocksdb_create_iterators(
 const rocksdb_snapshot_t* rocksdb_create_snapshot(
     rocksdb_t* db) {
   rocksdb_snapshot_t* result = new rocksdb_snapshot_t;
+  // if enable inplace_update_support, result->rep will be nullptr
   result->rep = db->rep->GetSnapshot();
   return result;
 }
 
 void rocksdb_release_snapshot(
     rocksdb_t* db,
     const rocksdb_snapshot_t* snapshot) {
-  db->rep->ReleaseSnapshot(snapshot->rep);
+  if (snapshot->rep != nullptr) {
```

Comment:
We need to keep the behavior of `rocksdb_release_snapshot()` and `DB::ReleaseSnapshot()` in sync. So adding a check for nullptr in the latter may be better.

btw, the decision of not adding a check there predates me. It would be nice if others can chime in and comment.

Reply:
ok, I agree with you.